File: 백준/Silver/2477. 참외밭/참외밭.py
count_melon = int(input())

max_height = 0
max_weight = 0
max_height_idx = 0
max_weight_idx = 0

location = [0 for _ in range(6)]
data = [0 for _ in range(6)]

for i in range(6) :
    line = input()
    values = list(map(int, line.split()))
    location[i] = values[0]
    data[i] = values[1]
    
for i in range(6) :
    if location[i] == 1 or location[i] == 2: #가로 
        if(data[i] > max_weight) :
            max_weight = data[i]
            max_weight_idx = i
    else :                                    # 세로
        if(data[i] > max_height) :
            max_height = data[i]
            max_height_idx = i
        
min_height = data[(max_height_idx+3)%6]
min_weight = data[(max_weight_idx+3)%6]

total = (max_height*max_weight) - (min_height*min_weight)

print(total*count_melon)

File: 백준/Silver/2477. 참외밭/참외밭.java
/**
 *  문제 이름 : 참외밭
 * 
 *  난 이  도 : 실버 2
 * 
 *  아이디어 : 참외의 갯수를 구하시오 (사실 상 면적 구하기 문제제)
 *            1: 동쪽, 2: 서쪽, 3: 남쪽, 4: 북쪽
 * 
 *  해    설 : 
 * 
 *  시간 복잡도 : 
 * 
 *  공간 복잡도 : 
 *    
 */

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {
    public static void main(String argc[]) throws IOException{
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter output = new BufferedWriter(new OutputStreamWriter(System.out));

        int fruits = Integer.parseInt(input.readLine());

        int array[] = new int[6];
        int side_array[] = new int[6];
        
        int max_height = 0;
        int max_height_index = 0;
        int max_weight = 0;
        int max_weight_index = 0;

        for(int i =0; i< 6; i++) {
            StringTokenizer token = new StringTokenizer(input.readLine());

            side_array[i] = Integer.parseInt(token.nextToken());
            array[i] = Integer.parseInt(token.nextToken());

            if(side_array[i] == 1 || side_array[i] == 2) {                
                if(max_height < array[i]) {
                    max_height = array[i];
                    max_height_index = i;
                }
            } else {
                if(max_weight < array[i]) {
                    max_weight = array[i];
                    max_weight_index = i;
                }
            }
        }

        int min_weight = Math.abs(array[(max_height_index+1)%6] - array[(max_height_index+5)%6]);
        int min_height = Math.abs(array[(max_weight_index+1)%6] - array[(max_weight_index+5)%6]);

        int result = (max_height * max_weight) - (min_height * min_weight);

        output.write(String.valueOf(result*fruits));

        input.close();
        output.close();
    }
}
